[bug] sync secrets failing

Run kumpeapps/kumpeapps-deployment-bot@main
Run CONFIG_PATH=""
Error: No deployment config found in .kumpeapps-deploy-bot/dev
Error: Process completed with exit code 1.

--- a/.github/workflows/sync-repository-secrets.yml
+++ b/.github/workflows/sync-repository-secrets.yml
@@ -17,16 +17,6 @@ name: Sync Repository Secrets (Reusable)
 on:
   workflow_call:
     inputs:
-      environment:
-        description: 'Environment (dev, stage, prod)'
-        required: false
-        type: string
-        default: 'dev'
-      config-path:
-        description: 'Path to deployment config (auto-detected if not provided)'
-        required: false
-        type: string
-        default: ''
       bot-url:
         description: 'Deployment bot URL (default: deploy.kumpe.app)'
         required: false
@@ -43,88 +33,38 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
-      
-      - name: Detect deployment config
-        id: config
-        run: |
-          CONFIG_PATH="${{ inputs.config-path }}"
-          
-          if [ -z "$CONFIG_PATH" ]; then
-            # Auto-detect: look for .kumpeapps-deploy-bot/{environment}/*.yml
-            ENV="${{ inputs.environment }}"
-            SEARCH_DIR=".kumpeapps-deploy-bot/${ENV}"
-            
-            if [ -d "$SEARCH_DIR" ]; then
-              # Find first .yml file in the directory
-              CONFIG_PATH=$(find "$SEARCH_DIR" -maxdepth 1 -name "*.yml" -o -name "*.yaml" | head -n1)
-            fi
-            
-            if [ -z "$CONFIG_PATH" ]; then
-              echo "❌ No deployment config found in $SEARCH_DIR"
-              exit 1
-            fi
-          fi
-          
-          echo "config_path=$CONFIG_PATH" >> $GITHUB_OUTPUT
-          echo "📄 Using config: $CONFIG_PATH"
-      
-      - name: Extract secrets from config
-        id: secrets
-        run: |
-          CONFIG_FILE="${{ steps.config.outputs.config_path }}"
-          
-          if [ ! -f "$CONFIG_FILE" ]; then
-            echo "❌ Config file not found: $CONFIG_FILE"
-            exit 1
-          fi
-          
-          # Extract env_mappings section and parse secret names
-          # Format: DATABASE_PASSWORD: DB_PASSWORD_SECRET
-          SECRET_NAMES=$(grep -A 20 'env_mappings:' "$CONFIG_FILE" | \
-            grep -E '^\s+[A-Z_]+:\s*[A-Z_]+' | \
-            awk '{print $2}' | \
-            sort -u | \
-            tr '\n' ',' | \
-            sed 's/,$//')
-          
-          if [ -z "$SECRET_NAMES" ]; then
-            echo "⚠️  No env_mappings found in $CONFIG_FILE"
-            echo "secret_names=" >> $GITHUB_OUTPUT
-            exit 0
-          fi
-          
-          echo "secret_names=$SECRET_NAMES" >> $GITHUB_OUTPUT
-          echo "🔑 Secrets to sync: $SECRET_NAMES"
-      
+
       - name: Sync secrets to deployment bot
-        if: steps.secrets.outputs.secret_names != ''
         env:
           BOT_URL: ${{ inputs.bot-url }}
           BOT_TOKEN: ${{ secrets.DEPLOYMENT_BOT_TOKEN }}
-          SECRET_NAMES: ${{ steps.secrets.outputs.secret_names }}
         run: |
           REPO_OWNER="${GITHUB_REPOSITORY%/*}"
           REPO_NAME="${GITHUB_REPOSITORY#*/}"
           
           echo "🚀 Syncing secrets for $REPO_OWNER/$REPO_NAME"
-          
-          # Convert comma-separated list to array
-          IFS=',' read -ra SECRETS <<< "$SECRET_NAMES"
+
+          # Collect env vars and filter out system/runtime variables
+          SECRET_NAMES=$(env | grep -E '^[A-Z][A-Z0-9_]*=' | cut -d= -f1 | grep -v -E '^(GITHUB_|RUNNER_|CI|PATH|HOME|USER|SHELL|PWD|OLDPWD|LANG|LC_|TERM|SHLVL|_|BOT_URL|BOT_TOKEN|ACTIONS_|AGENT_|ANDROID_|AZURE_|CHROME|CONDA|DEPLOYMENT_|DOTNET_|GOROOT|GRADLE_|GRAALVM_|HOMEBREW_|ImageOS|ImageVersion|INVOCATION_ID|JAVA_|JOURNAL_STREAM|LEIN_|NVM_|PERFLOG_|POWERSHELL_|SELENIUM_|SGX_|STATS_|SWIFT_|VCPKG_|INPUT_)')
+
+          if [ -z "$SECRET_NAMES" ]; then
+            echo "⚠️  No secrets found in environment variables to sync"
+            echo "⚠️  Make sure to pass your secrets via env in the caller workflow"
+            exit 0
+          fi
           
           SYNCED=0
           FAILED=0
           
-          for SECRET_NAME in "${SECRETS[@]}"; do
-            # Get secret value from GitHub secrets context
-            # Note: This requires secrets to be passed with 'secrets: inherit'
+          for SECRET_NAME in $SECRET_NAMES; do
+            if [ "$SECRET_NAME" = "KUMPEAPPS_DEPLOY_BOT_TOKEN" ] || [ "$SECRET_NAME" = "DEPLOYMENT_BOT_TOKEN" ]; then
+              continue
+            fi
+
             SECRET_VALUE=""
-            
-            # Try to get the secret value from environment
-            # GitHub Actions makes secrets available as env vars when explicitly referenced
             eval "SECRET_VALUE=\${${SECRET_NAME}:-}"
             
             if [ -z "$SECRET_VALUE" ]; then
-              echo "⚠️  Secret ${SECRET_NAME} not found in GitHub secrets - skipping"
               continue
             fi
             

--- a/README-ACTION.md
+++ b/README-ACTION.md
@@ -4,7 +4,7 @@ Automatically sync repository secrets from GitHub to the KumpeApps deployment bo
 
 ## Features
 
-- 📤 **Sync all secrets** - syncs everything you pass as environment variables
+- 🔑 **Sync all passed secrets** from workflow env variables
 - 🚀 **Simple setup** - just use the action, token is auto-provisioned
 - 🔒 **Secure** - uses per-repository API tokens, not admin tokens
 - 📝 **Audit trail** - all secret operations are logged
@@ -38,8 +38,7 @@ jobs:
       - uses: kumpeapps/kumpeapps-deployment-bot@v1
         env:
           KUMPEAPPS_DEPLOY_BOT_TOKEN: ${{ secrets.KUMPEAPPS_DEPLOY_BOT_TOKEN }}
-          # Pass all your secrets - they'll all be synced to the bot.
-          # Each deployment config uses only the secrets it needs.
+          # Pass all your secrets here - the action syncs everything provided
           DB_PASSWORD_SECRET: ${{ secrets.DB_PASSWORD_SECRET }}
           # Add your other secrets here
 ```
@@ -57,10 +56,10 @@ Go to Actions → Sync Secrets → Run workflow
    - Token only works for that specific repository (not admin access)
    - Provisioning runs at startup and hourly for resilience
 
-2. **Secret Sync** (all secrets):
-   - Action syncs ALL secrets passed as environment variables
-   - No config parsing - syncs everything you pass
-   - Simple and predictable behavior
+2. **Secret Sync** (automatic):
+  - Action reads all passed environment variables
+  - Filters out system/GitHub runtime variables
+  - Syncs remaining values as repository secrets in the bot
 
 3. **Deployment** (automatic):
    - Bot reads each deployment config's `env_mappings`
@@ -69,10 +68,14 @@ Go to Actions → Sync Secrets → Run workflow
 
 **Why do I need to list secrets in the workflow?**
 
+<<<<<<< HEAD
 GitHub Actions security prevents dynamic secret access (e.g., `${{ secrets[varName] }}`). The action syncs everything you pass:
 - 📤 Syncs ALL secrets passed as environment variables
 - 🎯 Each deployment config uses what it needs via `env_mappings`
 - 🔄 One workflow serves all environments
+=======
+GitHub Actions security prevents dynamic secret access (e.g., `${{ secrets[varName] }}`). The action syncs whatever secrets you explicitly pass in `env`.
+>>>>>>> 33daf68 ([bug/#16] Fixed Sync Secrets logic)
 
 ## Configuration
 
@@ -87,30 +90,19 @@ GitHub Actions security prevents dynamic secret access (e.g., `${{ secrets[varNa
 
 ```yaml
 jobs:
-  sync-secrets:
+  sync:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: kumpeapps/kumpeapps-deployment-bot@v1
         env:
           KUMPEAPPS_DEPLOY_BOT_TOKEN: ${{ secrets.KUMPEAPPS_DEPLOY_BOT_TOKEN }}
-          # Pass all your secrets - they'll all be synced
+          # Pass all your secrets - the action syncs everything provided
           DEV_DEPLOY_BOT_TOKEN: ${{ secrets.DEV_DEPLOY_BOT_TOKEN }}
           DEV_NEBULA_CLIENT_TOKEN: ${{ secrets.DEV_NEBULA_CLIENT_TOKEN }}
           PROD_DEPLOY_BOT_TOKEN: ${{ secrets.PROD_DEPLOY_BOT_TOKEN }}
           PROD_NEBULA_CLIENT_TOKEN: ${{ secrets.PROD_NEBULA_CLIENT_TOKEN }}
           DB_PASSWORD_SECRET: ${{ secrets.DB_PASSWORD_DEV }}
-  
-  sync-prod:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: kumpeapps/kumpeapps-deployment-bot@v1
-        with:
-          environment: prod
-        env:
-          KUMPEAPPS_DEPLOY_BOT_TOKEN: ${{ secrets.KUMPEAPPS_DEPLOY_BOT_TOKEN }}
-          DB_PASSWORD_SECRET: ${{ secrets.DB_PASSWORD_PROD }}
 ```
 
 ## Important Notes
@@ -119,14 +111,7 @@ jobs:
 
 Due to GitHub Actions security, you **must pass secrets as environment variables** in the workflow. The action cannot dynamically access `${{ secrets[varName] }}`.
 
-**However, the action is smart:**
-- 📖 Automatically reads your deployment config
-- ✓ Validates that required secrets are present
-- 📤 Only syncs secrets that are in your `env_mappings`
-- ⚠️ Warns if a required secret is missing
-- ⏭️ Skips secrets not needed by your config
-
-**Best practice:** List all your secrets in the workflow once, and the action handles the rest based on your deployment config.
+**Best practice:** Pass only the secrets you want stored for that repository.
 
 ### Token Security
 
@@ -139,7 +124,6 @@ Due to GitHub Actions security, you **must pass secrets as environment variables
 
 Sync secrets when:
 - ✅ Adding/updating secrets in GitHub
-- ✅ Changing `env_mappings` in deployment config
 - ✅ First-time repository setup
 
 You don't need to sync before every deployment - secrets are cached in the bot's database.
@@ -153,18 +137,16 @@ The `KUMPEAPPS_DEPLOY_BOT_TOKEN` secret hasn't been created. This should happen 
 2. Check repository Settings → Secrets → Actions for the token
 3. Check bot logs for token provisioning errors
 
-### "Secret XYZ not found"
+### "No secrets found in environment variables to sync"
 
-The action detected that your deployment config's `env_mappings` requires secret `XYZ`, but it wasn't passed as an environment variable.
+No custom secrets were passed in your workflow `env` block.
 
-**Fix:** Add it to your workflow's `env` section:
+**Fix:** Add secrets explicitly:
 ```yaml
 env:
   KUMPEAPPS_DEPLOY_BOT_TOKEN: ${{ secrets.KUMPEAPPS_DEPLOY_BOT_TOKEN }}
-  XYZ: ${{ secrets.XYZ }}  # Add this line
+  XYZ: ${{ secrets.XYZ }}
 ```
-
-The action automatically reads your deployment config to determine which secrets are needed, then validates they're all present.
 
 ### "Failed to sync secret (HTTP 401)"
 

--- a/action.yml
+++ b/action.yml
@@ -43,9 +43,7 @@ runs:
         echo ""
         
         # Get all environment variables, excluding system/GitHub Actions variables
-        # We'll sync anything that looks like a secret (uppercase with underscores)
         SECRET_NAMES=$(env | grep -E '^[A-Z][A-Z0-9_]*=' | cut -d= -f1 | grep -v -E '^(GITHUB_|RUNNER_|CI|PATH|HOME|USER|SHELL|PWD|OLDPWD|LANG|LC_|TERM|SHLVL|_|BOT_URL|BOT_TOKEN|ACTIONS_|AGENT_|ANDROID_|AZURE_|CHROME|CONDA|DEPLOYMENT_|DOTNET_|GOROOT|GRADLE_|GRAALVM_|HOMEBREW_|ImageOS|ImageVersion|INVOCATION_ID|JAVA_|JOURNAL_STREAM|LEIN_|NVM_|PERFLOG_|POWERSHELL_|SELENIUM_|SGX_|STATS_|SWIFT_|VCPKG_|INPUT_)')
-        
         if [ -z "$SECRET_NAMES" ]; then
           echo "::warning::No secrets found in environment variables to sync"
           echo "::warning::Make sure to pass your secrets in the env section of the action"
@@ -56,7 +54,6 @@ runs:
         FAILED=0
         
         for SECRET_NAME in $SECRET_NAMES; do
-          # Skip KUMPEAPPS_DEPLOY_BOT_TOKEN itself
           if [ "$SECRET_NAME" = "KUMPEAPPS_DEPLOY_BOT_TOKEN" ]; then
             continue
           fi

--- a/docs/secret-sync-guide.md
+++ b/docs/secret-sync-guide.md
@@ -55,22 +55,10 @@ See [docs/examples/repo-sync-secrets-simple.yml](./examples/repo-sync-secrets-si
 
 **Q: Why do I need to list secrets in the workflow?**
 
-GitHub Actions security prevents dynamic secret access (`${{ secrets[varName] }}`). **The action syncs everything you pass:**
+GitHub Actions security prevents dynamic secret access (`${{ secrets[varName] }}`).
+The action syncs the secrets you explicitly pass in the workflow `env` block.
 
-- 📤 Syncs ALL secrets passed as environment variables
-- 🎯 No config parsing - simple and predictable
-- 🔧 Each deployment config uses what it needs via `env_mappings`
-
-**You list all your secrets once** in the workflow, and:
-1. Action syncs all of them to the bot's database
-2. Each deployment config specifies which secrets it needs in `env_mappings`
-3. Bot injects only the secrets specified in that config's `env_mappings`
-
-**Example:**
-- You pass `DB_SECRET`, `API_SECRET`, and `NEBULA_TOKEN` in the workflow
-- All three are synced to the bot
-- Dev config has `env_mappings: { DB_PASS: DB_SECRET }` → gets only DB_SECRET
-- Prod config has `env_mappings: { DB_PASS: DB_SECRET, API_KEY: API_SECRET }` → gets both
+**You only need to list secrets once** in the workflow, and those values are synced to the bot.
 
 ### Step 3: Run the Workflow
 

